x-pack/plugins/alerting/server/application/rule/methods/delete/delete_rule.ts

```diff
@@ -5,40 +5,51 @@
  * 2.0.
  */
 
+import Boom from '@hapi/boom';
 import { AlertConsumers } from '@kbn/rule-data-utils';
-import { RawRule } from '../../types';
-import { WriteOperations, AlertingAuthorizationEntity } from '../../authorization';
-import { retryIfConflicts } from '../../lib/retry_if_conflicts';
-import { bulkMarkApiKeysForInvalidation } from '../../invalidate_pending_api_keys/bulk_mark_api_keys_for_invalidation';
-import { ruleAuditEvent, RuleAuditAction } from '../common/audit_events';
-import { RulesClientContext } from '../types';
-import { untrackRuleAlerts, migrateLegacyActions } from '../lib';
-import { RuleAttributes } from '../../data/rule/types';
-import { RULE_SAVED_OBJECT_TYPE } from '../../saved_objects';
+import { RawRule } from '../../../../types';
+import { WriteOperations, AlertingAuthorizationEntity } from '../../../../authorization';
+import { retryIfConflicts } from '../../../../lib/retry_if_conflicts';
+import { bulkMarkApiKeysForInvalidation } from '../../../../invalidate_pending_api_keys/bulk_mark_api_keys_for_invalidation';
+import { ruleAuditEvent, RuleAuditAction } from '../../../../rules_client/common/audit_events';
+import { RulesClientContext } from '../../../../rules_client/types';
+import { untrackRuleAlerts, migrateLegacyActions } from '../../../../rules_client/lib';
+import { RuleAttributes } from '../../../../data/rule/types';
+import { RULE_SAVED_OBJECT_TYPE } from '../../../../saved_objects';
+import { DeleteRuleParams } from './types';
+import { deleteRuleParamsSchema } from './schemas';
+import { deleteRuleSo, getDecryptedRuleSo, getRuleSo } from '../../../../data/rule';
+
+export async function deleteRule(context: RulesClientContext, params: DeleteRuleParams) {
+  try {
+    deleteRuleParamsSchema.validate(params);
+  } catch (error) {
+    throw Boom.badRequest(`Error validating delete params - ${error.message}`);
+  }
+
+  const { id } = params;
 
-export async function deleteRule(context: RulesClientContext, { id }: { id: string }) {
   return await retryIfConflicts(
     context.logger,
     `rulesClient.delete('${id}')`,
-    async () => await deleteWithOCC(context, { id })
+    async () => await deleteRuleWithOCC(context, { id })
   );
 }
 
-async function deleteWithOCC(context: RulesClientContext, { id }: { id: string }) {
+async function deleteRuleWithOCC(context: RulesClientContext, { id }: { id: string }) {
   let taskIdToRemove: string | undefined | null;
   let apiKeyToInvalidate: string | null = null;
   let apiKeyCreatedByUser: boolean | undefined | null = false;
-  let attributes: RawRule;
+  let attributes: RuleAttributes;
 
   try {
-    const decryptedAlert =
-      await context.encryptedSavedObjectsClient.getDecryptedAsInternalUser<RawRule>(
-        RULE_SAVED_OBJECT_TYPE,
-        id,
-        {
-          namespace: context.namespace,
-        }
-      );
+    const decryptedAlert = await getDecryptedRuleSo({
```

Comment:
nit: I assume these variables were named in the alert = rule era. If that's the case, should we take the occasion and update them to the new naming?